blockchain/vm/contracts.go

```diff
@@ -98,6 +99,59 @@ var PrecompiledContractsIstanbul = map[common.Address]PrecompiledContract{
 	common.BytesToAddress([]byte{3, 255}): &validateSender{},
 }
 
+// PrecompiledContractsKore contains the default set of pre-compiled Klaytn
+// contracts based on Ethereum Berlin.
+var PrecompiledContractsKore = map[common.Address]PrecompiledContract{
+	common.BytesToAddress([]byte{1}):      &ecrecover{},
+	common.BytesToAddress([]byte{2}):      &sha256hash{},
+	common.BytesToAddress([]byte{3}):      &ripemd160hash{},
+	common.BytesToAddress([]byte{4}):      &dataCopy{},
+	common.BytesToAddress([]byte{5}):      &bigModExp{eip2565: true},
+	common.BytesToAddress([]byte{6}):      &bn256AddIstanbul{},
+	common.BytesToAddress([]byte{7}):      &bn256ScalarMulIstanbul{},
+	common.BytesToAddress([]byte{8}):      &bn256PairingIstanbul{},
+	common.BytesToAddress([]byte{9}):      &blake2F{},
+	common.BytesToAddress([]byte{3, 253}): &vmLog{},
+	common.BytesToAddress([]byte{3, 254}): &feePayer{},
+	common.BytesToAddress([]byte{3, 255}): &validateSender{},
+}
+
+var (
+	PrecompiledAddressesKore                []common.Address
+	PrecompiledAddressesIstanbulCompatible  []common.Address
+	PrecompiledAddressesByzantiumCompatible []common.Address
+)
+
+func init() {
+	for k := range PrecompiledContractsByzantiumCompatible {
+		PrecompiledAddressesByzantiumCompatible = append(PrecompiledAddressesByzantiumCompatible, k)
+	}
+
+	// For istanbulCompatible, need to support for vmversion0 contracts, too. VmVersion0 contracts are deployed before istanbulCompatible.
+	// they use byzantiumCompatible precompiled contracts because they were deployed before istanbulCompatible.
```

Comment:
Does this make use byzantiumCompatible precompiled contracts after the istanbul hardfork as well?

Reply:
Yes, but only the contracts which are deployed before istanbulCompatible hf uses byzantiumCompatible precompiled contracts.

Reply:
@jeongkyun-oh oh, we have added `VmVersion` at istanbul HF. The related PR is https://github.com/klaytn/klaytn/pull/912.
If VmVersion is zero, it's deployed before istanbulCompatible HF.
If VmVersion is one, it's deployed after istanbulCompatible HF.

Reply:
@yoomee1313 Oh, yes. You're right. Thanks!

Reply:
@jeongkyun-oh Thanks for the nice comments.. it really helped a lot!